components/sync15/src/request.rs

```diff
@@ -1242,11 +1257,114 @@ mod test {
+        let cfg = InfoConfiguration {
+            max_post_bytes: 200,
+            max_total_bytes: 300,
+            max_post_records: 2,
+            max_total_records: 3,
+            ..InfoConfiguration::default()
+        };
+        let time = 11_111_111_000;
+        let (mut pq, tester) = pq_test_setup(
+            cfg,
+            time,
+            vec![
+                fake_response(status_codes::ACCEPTED, time, Some("1234")),
+                fake_response(status_codes::ACCEPTED, time + 100_000, Some("1234")), // should commit
+                fake_response(status_codes::ACCEPTED, time + 100_000, Some("abcd")),
+                fake_response(status_codes::ACCEPTED, time + 200_000, Some("abcd")), // should commit
+            ],
+        );
+
+        pq.enqueue(&make_record(100)).unwrap();
+        pq.enqueue(&make_record(100)).unwrap();
+        assert_eq!(pq.last_modified.0, time);
+        // POST
+        pq.enqueue(&make_record(100)).unwrap();
+        // POST + COMMIT
+        pq.enqueue(&make_record(100)).unwrap();
+        assert_eq!(pq.last_modified.0, time + 100_000);
+        pq.enqueue(&make_record(100)).unwrap();
+        assert_eq!(pq.last_modified.0, time + 100_000);
+        pq.flush(true).unwrap(); // COMMIT
+
+        assert_eq!(pq.last_modified.0, time + 200_000);
+
+        let t = tester.borrow();
+        assert!(t.cur_batch.is_none());
+        assert_eq!(t.all_posts.len(), 4);
+        assert_eq!(t.batches.len(), 2);
+        assert_eq!(t.batches[0].posts.len(), 2);
+        assert_eq!(t.batches[1].posts.len(), 1);
+
+        assert_eq!(t.batches[0].records, 3);
+        assert_eq!(t.batches[1].records, 2);
+
+        assert_eq!(t.batches[0].bytes, 300);
+        assert_eq!(t.batches[1].bytes, 200);
+
+        assert_eq!(t.batches[0].posts[0].batch.as_ref().unwrap(), "true");
+        assert_eq!(t.batches[0].posts[0].records, 2);
+        assert_eq!(t.batches[0].posts[0].payload_bytes, 200);
+        assert_eq!(t.batches[0].posts[0].commit, false);
+        assert_eq!(
+            t.batches[0].posts[0].body.len(),
+            request_bytes_for_payloads(&[100, 100])
+        );
+
+        assert_eq!(t.batches[0].posts[1].batch.as_ref().unwrap(), "1234");
+        assert_eq!(t.batches[0].posts[1].records, 1);
+        assert_eq!(t.batches[0].posts[1].payload_bytes, 100);
+        assert_eq!(t.batches[0].posts[1].commit, true);
+        assert_eq!(
+            t.batches[0].posts[1].body.len(),
+            request_bytes_for_payloads(&[100])
+        );
+
+        assert_eq!(t.batches[1].posts[0].batch.as_ref().unwrap(), "true");
+        assert_eq!(t.batches[1].posts[0].records, 2);
+        assert_eq!(t.batches[1].posts[0].payload_bytes, 200);
+        assert_eq!(t.batches[1].posts[0].commit, false);
+        assert_eq!(
+            t.batches[1].posts[0].body.len(),
+            request_bytes_for_payloads(&[100, 100])
+        );
+    }
+    
+    #[test]
+    fn test_error_cases() {
```

Comment:
Similarly here, is this WIP?

Reply:
Yes, this is WIP because each time I go back over the code I find a small warning or error catch that needs to be tested. I don't want to merge the PR until I test all error cases.

Reply:
👍